### PR TITLE
refactor events

### PR DESCRIPTION
**Pull request by iaroslavshvets**

### CODE REVIEW

File: packages/lib/src/createEventEmitter/createEventEmitter.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import {InnerEventEmitter} from './createEventEmitter.types';
+
+export const createEventEmitter = () => {
```
Comment (iaroslavshvets) [inline packages/lib/src/createEventEmitter/createEventEmitter.ts:3]: considering typings it's easier to write own impl for 40 rows, then trying to adopt some of solutions out there. Usually they are either untyped or require Record<'eventName', arguments> typings, which is leading to very complicated typings for `trigger` or passing some stuff like `undefined as any as never` as second argument when event doesn't pass anything to `trigger` because event have to be defined like 
```
type Events = {
 'someEvent:begin':  never;
 'someEvent:end: {errors: Errors}; 
}
```
TLDR:
it wasn't worth to fight these typings. Implementation is rather simple and straight-forward. Fully typed callback for every type of event.

File: packages/lib/src/createEventEmitter/createEventEmitter.types.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import type {Errors, SubmitParams} from '../FormControllerClass/FormControllerClass.types';
+
+export type FormagusEvent =
```
Comment (iaroslavshvets) [inline packages/lib/src/createEventEmitter/createEventEmitter.types.ts:3]: Just moved these types from another file (controller.types)

File: packages/lib/src/createEventEmitter/createEventEmitter.types.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      type: 'validateField:begin';
+      name: string;
+    }
+  | {
+      type: 'validateField:end';
+      name: string;
+      errors: Errors;
+    };
+
+export type InnerEventEmitter = {
```
Comment (iaroslavshvets) [inline packages/lib/src/createEventEmitter/createEventEmitter.types.ts:27]: Had to define it instead of ReturnType<typeof createEventEmitter> as it creates circular dependency

File: packages/lib/src/createEventEmitter/createEventEmitter.ts (hunk 1 above)
Comment (iaroslavshvets) [reply packages/lib/src/createEventEmitter/createEventEmitter.ts:4]: Found `strict-event-emitter`, which plays well with types
```
 type FormagusEvent = {
  'submit:begin': [],
  'submit:end': [Omit<SubmitParams, 'event'>],
}
  ```
and no issues with usage.
but it's pretty big and adds 5 kb more to bundle. I think, considering that this is pretty basic stuff - it's not worth it.

Comment (iaroslavshvets) [reply packages/lib/src/createEventEmitter/createEventEmitter.ts]: Wow. Found `mitt`. It's super small and well typed. Gonna use it. The only downside is that own impl returned unsubscribe function from `on` call, and `mitt` doesn't do it, you have to call `off` with exact function passed as listener to remove specific listener (like unsubscribe for DOM events)

Comment (iaroslavshvets) [reply packages/lib/src/createEventEmitter/createEventEmitter.ts]: see this commit 
https://github.com/iaroslavshvets/formagus/pull/105/commits/7f33595ff8222f385e58fc409e13c80a98585287
